gadget/install/partition.go

```diff
@@ -122,7 +122,11 @@ func createMissingPartitions(dl *gadget.OnDiskVolume, pv *gadget.LaidOutVolume,
 	}
 
 	// Make sure the devices for the partitions we created are available
-	if err := ensureNodesExist(created, 5*time.Second); err != nil {
+	ods := []gadget.OnDiskStructure{}
+	for _, ls := range created {
+		ods = append(ods, ls.OnDiskStructure)
+	}
+	if err := ensureNodesExist(ods, 5*time.Second); err != nil {
```

Comment:
given that we need to do the append copying, shouldn't we change ensureNodesExist to take a list of node strings?

Reply:
Changed now, that has also the benefit of simplifying some tests.